### ISSUE TITLE
Constructor returns @UnderInitialization object with @UnderInitialization argument

### ISSUE THREAD

**Issue by ky-gog** [typetools/checker-framework#5788]
Not sure if this is related to #409 but the setup is similar. 

When a constructor for a class `Bar` takes a `@UnderInitialization` object as argument, it's return type somehow becomes `@UnderInitialization(Bar.class) Bar`. This sometimes cause false positive errors.

# Command
`checker/bin/javac -processor nullness Foo.java`

# Input
File `Foo.java`
```java
final class Foo {
  private static class Bar {
    private Bar(@UnderInitialization Foo foo) {}
  }

  public Foo() {
    bar = new Bar(this);
  }

  private Bar bar;
}

```

# Output
```
Foo.java:12: error: [assignment] incompatible types in assignment.
    bar = new Bar(this);
          ^
  found   : @UnderInitialization(com.google.Foo.Bar.class) @NonNull Bar
  required: @Initialized @NonNull Bar
1 error
```

# Expectation

Return type of a constructor should always be `@Initialized`.
